aten/src/ATen/native/UpSampleNearest3d.cpp

```diff
@@ -3,145 +3,79 @@
 #include <ATen/native/UpSample.h>
 
 namespace at {
-namespace native {
-namespace {
-
-static void upsample_nearest3d_out_cpu_template(
-    Tensor& output,
-    const Tensor& input,
-    IntArrayRef output_size,
-    c10::optional<double> scales_d,
-    c10::optional<double> scales_h,
-    c10::optional<double> scales_w) {
+namespace meta {
+
+TORCH_META_FUNC(upsample_nearest3d) (
+  const Tensor& input,
+  IntArrayRef output_size,
+  c10::optional<double> scales_d,
+  c10::optional<double> scales_h,
+  c10::optional<double> scales_w
+) {
+  auto full_output_size = native::upsample_3d_common_check(input.sizes(), output_size);
+
+  // Allow for empty batch size but not other dimensions
   TORCH_CHECK(
-      output_size.size() == 3,
-      "It is expected output_size equals to 3, but got size ",
-      output_size.size());
-
-  int64_t output_depth = output_size[0];
-  int64_t output_height = output_size[1];
-  int64_t output_width = output_size[2];
-
-  int64_t nbatch = input.size(0);
-  int64_t channels = input.size(1);
-  int64_t input_depth = input.size(2);
-  int64_t input_height = input.size(3);
-  int64_t input_width = input.size(4);
-
-  upsample_3d_shape_check(
-      input,
-      Tensor(),
-      nbatch,
-      channels,
-      input_depth,
-      input_height,
-      input_width,
-      output_depth,
-      output_height,
-      output_width);
-
-  output.resize_({nbatch, channels, output_depth, output_height, output_width}, input.suggest_memory_format());
```

Comment:
@fmassa @bdhirsh This looks like a bug; previously `resize_` took the suggested memory format from input, but the call to `suggest_memory_format` is completely lost in the code below.

Reply:
cc @vfdev-5 can you try changing this locally and see if it fixes the behavior you were seeing?

Reply:
It looks like this bug also applies to upsample_nearest2d

Reply:
(but not upsample_nearest1d, which seems to have never gotten memory format support)

Reply:
Looks like we might need to audit the other places as well as for 2d this also seemed to happen https://github.com/pytorch/pytorch/pull/52012/files#r589563928

Reply:
Sorry about that :/ Happy to go through the PR's in this stack and fix that if someone hasn't started already

Reply:
@bdhirsh if you could fix it it would be great

Reply:
Additional wrinkle here is that cuda seems to not support layout propagation for this op, and backward layout propagation is not supported neither for cuda nor cpu (unless I'm overlooking something).

Reply:
leaving the fix here for the paper trail: https://github.com/pytorch/pytorch/pull/53535/